restdocs-openapi-gradle-plugin/src/main/kotlin/com/epages/restdocs/openapi/gradle/OpenApi20Generator.kt

```diff
@@ -59,45 +61,49 @@ internal object OpenApi20Generator {
     fun extractDefinitions(swagger: Swagger) : Swagger {
         val schemasToKeys = HashMap<Model, String>()
 
-        swagger.paths.values.forEach {
-            it.operations.forEach {
-                val parameter = it.parameters
-                    .filter {
-                        it.`in`.equals("body")
+        swagger.paths.values.map { it.operations }.flatten().forEach {
+            extractBodyParameter(it.parameters)
+                    .let {
+                        if(it != null && it.schema != null) {
+                            it.schema(extractOrFindSchema(schemasToKeys, it.schema, generateSchemaName()))
+                        }
                     }
-                    .map { it as BodyParameter }
-                    .firstOrNull()
-                //TODO response schema
-
-                if(parameter != null && parameter.schema != null) {
-                    val key = extractOrFindSchema(schemasToKeys, parameter.schema)
-                    val model = RefModel("#/definitions/$key")
-                    parameter.setSchema(model)
-                }
-            }
-        }
-
-        val definitions = HashMap<String, Model>()
 
-        schemasToKeys.keys.forEach {
-            val schemaName = schemasToKeys.getValue(it)
-            definitions.put(schemaName, it)
+            it.responses.values
+                .filter { it.responseSchema != null }
+                .forEach {
+                    it.responseSchema(extractOrFindSchema(schemasToKeys, it.responseSchema, generateSchemaName()))
+                }
         }
 
-        swagger.setDefinitions(definitions)
+        swagger.definitions =
+            schemasToKeys.keys.map {
+                schemasToKeys.getValue(it) to it
+            }.toMap()
 
         return swagger
     }
 
-    private fun extractOrFindSchema(schemasToKeys: HashMap<Model, String>, schema: Model): String {
-        if (schemasToKeys.containsKey(schema)) {
-            return schemasToKeys.get(schema)!!
+    private fun extractBodyParameter(parameters: List<Parameter>): BodyParameter? {
+        return parameters
+            .filter { it.`in`.equals("body") }
+            .map { it as BodyParameter }
+            .firstOrNull()
+    }
+
+    private fun extractOrFindSchema(schemasToKeys: HashMap<Model, String>, schema: Model, schemaName: String): Model {
```

Comment:
Right now we are generating a schemaName every time we call this function - regardless whether it's needed or not.
One way to fix this: Instead of passing the concrete name, pass a generator function that could generate a name if it is needed.

But imo the schemaName or the generator function does not need to be passed as a parameter at all.
The schema name can be generated locally in this function, esp. in the future if we consider the schema fields.